blobspike/README.md

```diff
@@ -0,0 +1,198 @@
+# Azure Blob Storage Copy
+
+We evaluated the possibility to copy a blob.
+
+## Copy with Java SDK
+
+We evaluated the [Java azure storage client library](https://docs.microsoft.com/en-us/java/api/overview/azure/storage-blob-readme?view=azure-java-stable) to copy blobs.
+
+### Evaluate if copy can be done from source to destination, with the client not handling any data flow
+
+The Java Azure Storage Blob client library is on top of the [Blob service REST API](https://docs.microsoft.com/en-us/rest/api/storageservices/blob-service-rest-api).
+The client calls [an endpoint](https://docs.microsoft.com/en-us/rest/api/storageservices/copy-blob-from-url) to trigger the copy of a blob.
+The client receives an answer, the copy is then done asynchronously.
+In order to know if the copy is finished, the client needs to [get the blob properties](https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob-properties) by calling the REST API in order to know if the copy operation is finished.
+
+### Copy duration
+
+We tested copy between 2 containers in different tenants. The source blob was in West Europe and the destination in East US.
+
+- 1GB: 9 minutes.
+- 256MB: 30 to 40 seconds.
+- 1MB: 1 to 1.5 seconds.
+
+## Evaluation of copy vs streaming
+
+We have 2 solutions to transfer data from a blob to another blob.
+
+- Use the [BlobClient](https://docs.microsoft.com/en-us/java/api/com.azure.storage.blob.blobclient?view=azure-java-stable) beginCopy to trigger a copy of source blob to destination blob.
+- Use the [BlockBlobClient](https://docs.microsoft.com/en-us/java/api/com.azure.storage.blob.specialized.blockblobclient?view=azure-java-stable) to get a stream from blob to VM and transfer from VM to blob with another stream.
+
+Pros and cons to use beginCopy in EDC:
+- Pros: During the copy, Network I/O is only between the 2 storage accounts. The client does not need to download any data.
+- Cons: The source storage account needs to be accessible from the internet. With the stream, the provider and the source storage account could be in the same private network. Then, only the destination storage account would need to be accessible from the internet.
+
+### Azure Blob Copy within same container
```

Comment:
Is this related to "Evaluation of copy vs streaming"? Why is it a subsection of it?